niceman/distributions/conda.py

```diff
@@ -92,13 +92,43 @@ def install_packages(self, session=None):
             Environment sub-class instance.
         """
 
-        # TODO: Need to figure out a graceful way to install conda before we can install packages here.
-        # for package in self.provenance['packages']:
-        #     session.add_command(['conda',
-        #                            'install',
-        #                            package['name']])
+        # TODO: Install Conda
+#        print("We got" + str(environment))
+        # Loop through non-root packages, creating the conda-env export
+        # file to import
+        for env in self.environments:
+            if env.path == self.path:
+                continue
+            export_contents = self.create_conda_export(env)
+            print ("EXPORT\n" + export_contents)
+
         return
 
+    @staticmethod
+    def create_conda_export(env):
+        # TODO: The environment name should be discovered on retrace
+        name = os.path.basename(os.path.normpath(env.path))
+        export = "name: %s\n" % name
+        # Collect channels
+        export += "channels:\n"
+        for c in env.channels:
+            export += "- %s\n" % c["name"]
+        export += "dependencies:\n"
+        # Collect packages (separating pip installed packages)
+        pip_export = ""
+        for p in env.packages:
+            if p.get("installer") is None:
```

Comment:
I think you'd more commonly see this check as `"installer" in p` or as
`p["install"]` with a `KeyError` catch.

Reply:
I made the change but then I realized: "installer" actually is in p, but set to None...  So I'm not checking for existence but actual definition (we have a number of mandatory fields in our objects that can be set to None).

Reply:
Ah, sorry.  Should have looked more closely.  Since it's mandatory, you can just use the getitem access without worrying about a KeyError then, right?